### DISCUSSION TITLE
How to use this in python?

### DISCUSSION THREAD

**Discussion by jaytnw** [jblance/mpp-solar#101] [Q&A]
I want use it in my python code for create my app, show data and setting my inverter , How to use it ?

**Comment by jblance** [comment]
You'll need to be more specific
It is not really setup for app usage - but my guess is that you need to import the relevant modules into your code and drive them using gui triggers

**Reply by lwjhussein** [reply]
Hi I have installed a new raspberry pi 4 and tried to follow the commands. If i use pip i get errors. if i use pip3 its seams to install something but i am un able to proceed. The guidance seems to assume some prior knowledge on how to install?

**Reply by jblance** [reply]
Yes it does require some degree of knowledge. What happens have you install with pip3? (Post the output) what are you trying to do? (What is connected to the pi, how is it connected, where is it connected etc)

**Comment by lwjhussein** [comment]
Thank you for the reply.
When i then put pip3 on the commands it has installed. 
Now still my graphs are blank though. 
When i run the command
mpp-solar -p /dev/hidraw0 -c QPI -o mqtt -q mqttbroker
I get:

WARNING:mqtt:output: Error publishing MQTT messages to broker 'mqttbroker' on port '1883' with auth 'None'WARNING:mqtt:[Errno -2] Name or service not ***@***.***:~ $
Would you have an image with all the basics set up that i can try?
Regards

Sent from Yahoo Mail on Android 
 
  On Fri, May 21, 2021 at 22:54, John ***@***.***> wrote:   


Yes it does require some degree of knowledge. What happens have you install with pip3? (Post the output) what are you trying to do? (What is connected to the pi, how is it connected, where is it connected etc)

**Reply by jblance** [reply]
No preset up images as everyone's setup is different 

1. What inverter do you have?
2. Start with getting it to display to the screen (remove the `-o mqtt` )
3. Do you have a mqtt broker running at the address `mqttbroker`
4. What graphs are you trying to populate?

**Comment by lwjhussein** [comment]
Thanks again.
I have a Synapse 5kw (axpert King) inverter connected to 3  lithium batteries, and 12 300w panels.
I would like to show the status of the power input, output. Basically to be able to remote monitor the inverter and power status. I use duckdns to connect to my home network. 
To get to this stage i have used Google and manager to resolve some errors but more a wits end. Not sure if i should start a fresh install. 

 I have imported this template into Grafana but its currently blank. PIP5048MG.json dashboard for Grafana  
|  
|   
|   
|   |    |

   |

  |
|  
|   |  
PIP5048MG.json dashboard for Grafana
 
Dashboard for remote monitoring mppsolar Inverters
  |   |

  |

  |

  




Sent from Yahoo Mail on Android 
 
  On Sat, May 22, 2021 at 8:46, John ***@***.***> wrote:   


No preset up images as everyone's setup is different
   
   - What inverter do you have?
   - Start with getting it to display to the screen (remove the -o mqtt )
   - Do you have a mqtt broker running at the address mqttbroker
   - What graphs are you trying to populate?

**Reply by jblance** [reply]
What do you get when you run
`mpp-solar -p /dev/hidraw0 -c QPI -I`
Post all the output

**Comment by lwjhussein** [comment]
JBlance
Thank you for the assistance
Here is the output.

```
~ $ mpp-solar -p /dev/hidraw0 -c QPI -I
INFO:root:Solar Device Command Utility, version: 0.7.49, recent changes: add DALY cellVoltages
INFO:root:Creating device "unnamed" (type: "mppsolar") on port "/dev/hidraw0 (porttype=None)" using protocol "PI30"
INFO:device:set_port: Using hidrawio for communications
INFO:root:Looping 1 commandsINFO:root:Getting results from device: mppsolar device - name: unnamed, port: <mppsolar.io.hidrawio.HIDRawIO object at 0xb654e9f0>, protocol: <mppsolar.protocols.pi30.pi30 object at 0xb654eb90> for command: QPI, tag: QPI, outputs: screen
INFO:device:run_command: Running command QPI
INFO:AbstractProtocol:get_full_command: Using protocol b'PI30' with 36 commands
INFO:device:run_command: full command b'QPI\xbe\xac\r' for command QPI
INFO:AbstractProtocol:decode: response passed to decode: b'(PI30\x9a\x0b\r'
INFO:AbstractProtocol:decode: Processing response of type DEFAULT
INFO:AbstractProtocol:decode: Processing DEFAULT type responses
INFO:device:run_command: Decoded esponse {'raw_response': ['(PI30\x9a\x0b\r', ''], '_command': 'QPI', '_command_description': 'Protocol ID inquiry', 'Protocol ID': ['PI30', '']}
INFO:root:attempting to create output processor: screen
INFO:screen:output: Using output processor: screen
Command: QPI - Protocol ID inquiry
------------------------------------------------------------
Parameter           Value      Unit
protocol_id
```

**Reply by jblance** [reply]
reformated for clarity.

Looks like it is working (though the content you posted didnt show the actual response) on the screen it should look like:
```
Command: QPI - Protocol ID inquiry
------------------------------------------------------------
Parameter           Value      Unit
protocol_id          PI30
```

Show now try with `mpp-solar -p /dev/hidraw0 -c QPIGS -I`

**Comment by lwjhussein** [comment]
Hi again, the format is from copying the content remotely.

The latest out put is
***@***.***:~ $ mpp-solar -p /dev/hidraw0 -c QPIGS -IINFO:root:Solar Device Command Utility, version: 0.7.49, recent changes: add DALY cellVoltagesINFO:root:Creating device "unnamed" (type: "mppsolar") on port "/dev/hidraw0 (porttype=None)" using protocol "PI30"INFO:device:set_port: Using hidrawio for communicationsINFO:root:Looping 1 commandsINFO:root:Getting results from device: mppsolar device - name: unnamed, port: <mppsolar.io.hidrawio.HIDRawIO object at 0xb6521a10>, protocol: <mppsolar.protocols.pi30.pi30 object at 0xb6521bb0> for command: QPIGS, tag: QPIGS, outputs: screenINFO:device:run_command: Running command QPIGSINFO:AbstractProtocol:get_full_command: Using protocol b'PI30' with 36 commandsINFO:device:run_command: full command b'QPIGS\xb7\xa9\r' for command QPIGSINFO:AbstractProtocol:decode: response passed to decode: b' 113.0 51.20 00000 00010111 00 00 00071 110\xbe\x06\r'INFO:AbstractProtocol:Invalid response CRCINFO:device:run_command: Decoded response {'ERROR': ['Invalid response CRC', '']}INFO:root:attempting to create output processor: screenINFO:screen:output: Using output processor: screenParameter Value Uniterror Invalid response CRC

Sent from Yahoo Mail on Android 
 
  On Sat, May 22, 2021 at 12:04, John ***@***.***> wrote:   


reformated for clarity.

Looks like it is working (though the content you posted didnt show the actual response) on the screen it should look like:
Command: QPI - Protocol ID inquiry
------------------------------------------------------------
Parameter           Value      Unit
protocol_id          PI30

Show now try with mpp-solar -p /dev/hidraw0 -c QPIGS -I

**Comment by lwjhussein** [comment]
Herewith a ociture from running the same command while on the LAN.


Hi again, the format is from copying the content remotely.

The latest out put is
```
***@***.***:~ $ mpp-solar -p /dev/hidraw0 -c QPIGS -I
INFO:root:Solar Device Command Utility, version: 0.7.49, recent changes: add DALY cellVoltages
INFO:root:Creating device "unnamed" (type: "mppsolar") on port "/dev/hidraw0 (porttype=None)" using protocol "PI30"
INFO:device:set_port: Using hidrawio for communications
INFO:root:Looping 1 commands
INFO:root:Getting results from device: mppsolar device - name: unnamed, port: <mppsolar.io.hidrawio.HIDRawIO object at 0xb6521a10>, protocol: <mppsolar.protocols.pi30.pi30 object at 0xb6521bb0> for command: QPIGS, tag: QPIGS, outputs: screen
INFO:device:run_command: Running command QPIGS
INFO:AbstractProtocol:get_full_command: Using protocol b'PI30' with 36 commands
INFO:device:run_command: full command b'QPIGS\xb7\xa9\r' for command QPIGS
INFO:AbstractProtocol:decode: response passed to decode: b' 113.0 51.20 00000 00010111 00 00 00071 110\xbe\x06\r'
INFO:AbstractProtocol:Invalid response CRC
INFO:device:run_command: Decoded response {'ERROR': ['Invalid response CRC', '']}
INFO:root:attempting to create output processor: screen
INFO:screen:output: Using output processor: screenParameter Value Uniterror Invalid response CRC

```

**Reply by jblance** [reply]
> INFO:AbstractProtocol:decode: response passed to decode: b' 113.0 51.20 00000 00010111 00 00 00071 110\xbe\x06\r'

This is an incomplete response - is anything else talking to the inverter or using the serial port?

**Comment by lwjhussein** [comment]
Sorry been away. I was tring out pd-axpert. I have uninstalled it. 
New screen is show below 





Screen now has

Sent from Yahoo Mail on Android 
 
  On Fri, May 28, 2021 at 0:06, John ***@***.***> wrote:   



INFO:AbstractProtocol:decode: response passed to decode: b' 113.0 51.20 00000 00010111 00 00 00071 110\xbe\x06\r'


This is an incomplete response - is anything else talking to the inverter or using the serial port?

**Reply by jblance** [reply]
I dont see any screenshot?

**Comment by lwjhussein** [comment]
It appears set up as some of the commands listed under help.
The grafana databases are not picking anything. Uner ip:3000 i selected influxdb and out credentials and it gave a green tick.


***@***.***:~ $ mpp-solar -p /dev/hidraw0 -c QPIRICommand: QPIRI - Current Settings inquiry------------------------------------------------------------Parameter Value Unitac_input_voltage 230.0 Vac_input_current 21.7 Aac_output_voltage 230.0 Vac_output_frequency 50.0 Hzac_output_current 21.7 Aac_output_apparent_power 5000 VAac_output_active_power 5000 Wbattery_voltage 48.0 Vbattery_recharge_voltage 44.0 Vbattery_under_voltage 42.0 Vbattery_bulk_charge_voltage 54.0 Vbattery_float_charge_voltage 51.2 Vbattery_type Usermax_ac_charging_current 40 Amax_charging_current 50 Ainput_voltage_range UPSoutput_source_priority Solar firstcharger_source_priority Solar firstmax_parallel_units 9 unitsmachine_type Off Gridtopology transformerlessoutput_mode single machine outputbattery_redischarge_voltage 54.0 Vpv_ok_condition As long as one unit of inverters has connect PV, parallel system will consider PV OKpv_power_balance PV input max power will be the sum of the max charged power and loads powerunknown_value_in_response_25 000unknown_value_in_response_26 ***@***.***:~ $ mpp-solar -p /dev/hidraw0 -c QPIWSCommand: QPIWS - Warning status inquiry------------------------------------------------------------Parameter Value Unitwarning_status        - ***@***.***:~ $ mpp-solar -p /dev/hidraw0 -c QVFWCommand: QVFW - Main CPU firmware version inquiry------------------------------------------------------------Parameter Value Unitmain_cpu_firmware_version ***@***.***:~ $ mpp-solar -p /dev/hidraw0 -c QVFW2Command: QVFW2 - Secondary CPU firmware version inquiry------------------------------------------------------------Parameter Value Unitsecondary_cpu_firmware_version VERFW2:00000.00

Sent from Yahoo Mail on Android 
 
  On Sun, May 30, 2021 at 22:58, John ***@***.***> wrote:   


I dont see any screenshot?

**Comment by lwjhussein** [comment]
My current grafana pages




Sent from Yahoo Mail on Android 
 
  On Sun, May 30, 2021 at 23:29, lewis ***@***.***> wrote:   It appears set up as some of the commands listed under help.
The grafana databases are not picking anything. Uner ip:3000 i selected influxdb and out credentials and it gave a green tick.


***@***.***:~ $ mpp-solar -p /dev/hidraw0 -c QPIRICommand: QPIRI - Current Settings inquiry------------------------------------------------------------Parameter Value Unitac_input_voltage 230.0 Vac_input_current 21.7 Aac_output_voltage 230.0 Vac_output_frequency 50.0 Hzac_output_current 21.7 Aac_output_apparent_power 5000 VAac_output_active_power 5000 Wbattery_voltage 48.0 Vbattery_recharge_voltage 44.0 Vbattery_under_voltage 42.0 Vbattery_bulk_charge_voltage 54.0 Vbattery_float_charge_voltage 51.2 Vbattery_type Usermax_ac_charging_current 40 Amax_charging_current 50 Ainput_voltage_range UPSoutput_source_priority Solar firstcharger_source_priority Solar firstmax_parallel_units 9 unitsmachine_type Off Gridtopology transformerlessoutput_mode single machine outputbattery_redischarge_voltage 54.0 Vpv_ok_condition As long as one unit of inverters has connect PV, parallel system will consider PV OKpv_power_balance PV input max power will be the sum of the max charged power and loads powerunknown_value_in_response_25 000unknown_value_in_response_26 ***@***.***:~ $ mpp-solar -p /dev/hidraw0 -c QPIWSCommand: QPIWS - Warning status inquiry------------------------------------------------------------Parameter Value Unitwarning_status        - ***@***.***:~ $ mpp-solar -p /dev/hidraw0 -c QVFWCommand: QVFW - Main CPU firmware version inquiry------------------------------------------------------------Parameter Value Unitmain_cpu_firmware_version ***@***.***:~ $ mpp-solar -p /dev/hidraw0 -c QVFW2Command: QVFW2 - Secondary CPU firmware version inquiry------------------------------------------------------------Parameter Value Unitsecondary_cpu_firmware_version VERFW2:00000.00

Sent from Yahoo Mail on Android 
 
  On Sun, May 30, 2021 at 22:58, John ***@***.***> wrote:   


I dont see any screenshot?
